commit e993e21bf71d0ae306faa5f9333392171b7546a9 (refs/changes/32/2132/3)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-06-21T18:25:16+00:00 (2 years, 4 months ago)
    
    VOLAPI-54 Custom errors' rest codes should follow engineering guide's guidelines

diff --git a/lib/errors.js b/lib/errors.js
index cc52e8c..f5f678f 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -14,7 +14,7 @@ var util = require('util');
 
 function VolumeNotFoundError(volumeUuid) {
     restify.RestError.call(this, {
-        restCode: 'VOLUME_NOT_FOUND',
+        restCode: 'VolumeNotFound',
         statusCode: 404,
         message: 'Volume with uuid ' + volumeUuid + ' could not be found',
         constructorOpt: VolumeNotFoundError
@@ -25,7 +25,7 @@ util.inherits(VolumeNotFoundError, restify.RestError);
 
 function VolumeAlreadyExistsError(volumeName) {
     restify.RestError.call(this, {
-        restCode: 'VOLUME_ALREADY_EXISTS',
+        restCode: 'VolumeAlreadyExists',
         statusCode: 409,
         message: 'Volume with name ' + volumeName + ' already exists',
         constructorOpt: VolumeAlreadyExistsError
@@ -36,7 +36,7 @@ util.inherits(VolumeAlreadyExistsError, restify.RestError);
 
 function VolumeInUseError(volumeName) {
     restify.RestError.call(this, {
-        restCode: 'VOLUME_IN_USE',
+        restCode: 'VolumeInUse',
         statusCode: 409,
         message: 'Volume with name ' + volumeName + ' is used',
         constructorOpt: VolumeInUseError
@@ -49,7 +49,7 @@ function ValidationError(causes) {
     assert.arrayOfObject(causes, 'causes');
 
     restify.RestError.call(this, {
-        restCode: 'VALIDATION_ERROR',
+        restCode: 'ValidationError',
         statusCode: 409,
         message: 'Validation error, causes: ' + causes,
         constructorOpt: ValidationError
